spec/hooks/useGroups/useGroups.test.js

```diff
@@ -0,0 +1,281 @@
+import '@testing-library/jest-dom';
+import { renderHook, waitFor } from '@testing-library/react';
+import useGroups from '../../../src/hooks/useGroups';
+import mockSearchResponse from '../../local_examples/apiSearchResponse.json';
+import { transformSearchResponse } from '../../../src/utils/transformers';
+import { renderHookWithCioPlp } from '../../test-utils';
+
+describe('Testing Hook: useGroups', () => {
+  const originalWindowLocation = window.location;
+  const testGroupA = 'W676714'; // Clearance
+  const testGroupB = 'W127085'; // Men
+  const currentGroup = 'AP01';
+
+  beforeEach(() => {
+    // Mock console error to de-clutter the console for expected errors
+    const spy = jest.spyOn(console, 'error');
+    spy.mockImplementation(() => {});
+
+    Object.defineProperty(window, 'location', {
+      value: new URL('https://example.com'),
+    });
+  });
+
+  afterEach(() => {
+    Object.defineProperty(window, 'location', {
+      value: originalWindowLocation,
+    });
+    jest.restoreAllMocks(); // This will reset all mocks after each test
+  });
+
+  const searchData = transformSearchResponse(mockSearchResponse);
+  const useGroupsProps = { groups: searchData.response.groups };
+
+  it('Should throw error if called outside of PlpContext', () => {
+    expect(() => renderHook(() => useGroups())).toThrow();
+  });
+
+  it('Should return groups array', async () => {
+    const { result } = renderHookWithCioPlp(() => useGroups(useGroupsProps));
+
+    await waitFor(() => {
+      const {
+        current: { groups },
+      } = result;
+
+      expect(groups).toHaveLength(searchData.response.groups.length);
+      expect(groups).toEqual(searchData.response.groups);
+    });
+  });
+
+  it('setGroup should apply group filter correctly', async () => {
```

Comment:
I don't know what the expected behavior is but is Groups filter supposed to support multiple selects?

Reply:
It can, but for this version, since we're focusing on hierarchy navigation, I've scoped it out because this PR was getting way too big. Applying multiple groups as filters at the same time also seems like an edge case, so I don't think we need it until it's requested.